### PR TITLE
docs(examples): add nginx example chart

### PR DESCRIPTION
**Pull request by technosophos**
This adds an nginx chart that shows off more of the template system's
capabilities.

### CODE REVIEW

**Comment by adamreese** [comment]
wicked

File: docs/examples/nginx/README.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+This Helm chart provides examples of some of Helm's more powerful
+features.
+
+**This is not a production-grade chart. It is an example.**
+
+The chart installs a simple nginx server according to the following
+pattern:
+
+- A `ConfigMap` is used to store the files the server will serve.
+  (`templates/configmap.yaml`)
```
Comment (errordeveloper) [inline docs/examples/nginx/README.md]: Would be nice to have these as links.

File: docs/examples/nginx/README.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+- A `Deployment` is used to create a Replica Set of nginx pods.
+  (`templates/deployment.yaml`)
+- A `Service` is used to create a gateway to the pods running in the
+  replica set (`templates/svc.yaml`)
+
+The `values.yaml` exposes a few of the configuration options in the
+charts, though there are some that are not exposed there (like
+`.image`).
+
+The `templates/_helpers.tpl` file contains helper templates. The leading
```
Comment (errordeveloper) [inline docs/examples/nginx/README.md]: This makes me wonder if user create some number of such files, and if so why would the need to?

File: docs/examples/nginx/templates/deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    release: {{.Release.Name}}
+    # This makes it easy to audit chart usage.
+    chart: {{.Chart.Name}}-{{.Chart.Version}}
+spec:
+  replicas: {{default 1 .replicaCount}}
+  template:
+    metadata:
+      labels:
+        app: {{template "fullname" .}}
+        release: {{.Release.Name}}
```
Comment (errordeveloper) [inline docs/examples/nginx/templates/deployment.yaml]: Naturally, a use will likely read `docs/examples/alpine` first, and the [metadata there](https://github.com/kubernetes/helm/blob/0acbcdd5ccaceedf26871b416e2d775bf6e1c026/docs/examples/alpine/templates/alpine-pod.yaml#L3-L10) differs from this. Would be great to clarify what user really must supply.

File: docs/examples/nginx/templates/deployment.yaml [hunk 4]
```diff
@@ -0,0 +1,39 @@
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  # This uses a "fullname" template (see _helpers)
+  # Basing names on .Release.Name means that the same chart can be installed
+  # multiple times into the same namespace.
+  name: {{template "fullname" .}}
+  labels:
+    heritage: helm
```
Comment (errordeveloper) [inline docs/examples/nginx/templates/deployment.yaml]: Is this required?

**Comment by errordeveloper** [comment]
Great example! Few comments from me, as someone new to Helm :)

**Comment by errordeveloper** [comment]
Also, `name: {{template "fullname" .}}` is repeated a lot. Had there been any discussion of a shortcut for such thing?

Would be nice to say it once and later refer to is with some shorthand sugar. I would imagine that such a thing can be done with YAML pointers, if we care about YAML syntax validity, but not sure how it would fit...

**Comment by technosophos** [comment]
My main concern with using the more sophisticated YAML features is that I don't know how broad the support for those is. Honestly, I haven't even tested candiedyaml to see how much of v2 it supports. But I am all for shorthand where possible.

**Pull request merged**